Assets/MixedRealityToolkit.Tests/PlayModeTests/ManipulationHandlerTests.cs

```diff
@@ -518,48 +535,89 @@ public IEnumerator ManipulationHandlerOneHandRotateWithConstraint()
+            diffAngle = Quaternion.Angle(testObject.transform.rotation, initialQuaternion);
+            Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on Z axis did not lock axis correctly (world space)");
+
+            /*********************************/
+            /*** TEST LOCAL SPACE ROTATION ***/
+            /*********************************/
+
+            manipHandler.UseLocalSpaceForConstraint = true;
+            // rotate with x axis only
+            manipHandler.ConstraintOnRotation = RotationConstraintType.XAxisOnly;
+            yield return hand.SetRotation(testQuaternion, numRotSteps);
+            diffAngle = Quaternion.Angle(testObject.transform.localRotation, Quaternion.Euler(testRotation, 0, 0));
+            Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on x axis did not lock axis correctly (local space)");
+
+            yield return hand.SetRotation(Quaternion.identity, numRotSteps);
+            diffAngle = Quaternion.Angle(testObject.transform.localRotation, Quaternion.identity);
+            Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on x axis did not lock axis correctly (local space)");
+
+            // rotate with y axis only
+            manipHandler.ConstraintOnRotation = RotationConstraintType.YAxisOnly;
+            yield return hand.SetRotation(testQuaternion, numRotSteps);
+            diffAngle = Quaternion.Angle(testObject.transform.localRotation, Quaternion.Euler(0, testRotation, 0));
+            Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on Y axis did not lock axis correctly (local space)");
+
+            yield return hand.SetRotation(Quaternion.identity, numRotSteps);
+            diffAngle = Quaternion.Angle(testObject.transform.localRotation, Quaternion.identity);
+            Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on Y axis did not lock axis correctly (local space)");
+
+            // rotate with z axis only
+            manipHandler.ConstraintOnRotation = RotationConstraintType.ZAxisOnly;
+            yield return hand.SetRotation(testQuaternion, numRotSteps);
+            diffAngle = Quaternion.Angle(testObject.transform.localRotation, Quaternion.Euler(0, 0, testRotation));
+            Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on Z axis did not lock axis correctly (local space)");
+
+            yield return hand.SetRotation(Quaternion.identity, numRotSteps);
+            diffAngle = Quaternion.Angle(testObject.transform.localRotation, Quaternion.identity);
+            Assert.IsTrue(Mathf.Approximately(diffAngle, 0.0f), "constraint on Z axis did not lock axis correctly (local space)");
 
             yield return hand.SetGesture(ArticulatedHandPose.GestureId.Open);
             yield return hand.Hide();
+
+            UnityEngine.Object.Destroy(parentObject);
```

Comment:
Do you still need this, if you are doing the cleanup action above?

Reply:
Well, this will cleanup immediately after this method completes, as opposed to waiting for all tests to clean up the GameObject. Basically on failure I want clean up, on success I want cleanup as fast as possible.